docs/testsuite_new_language.md

```diff
@@ -0,0 +1,179 @@
+    For Python and Java, this means running a ```gradle build```. In Java, the code is generated in ```src/```. In Python, 
+    the code is generated  in ```build/```
+    
+    The generated code will be tested later in the testsuite.   
+    
+3.  Select a test framework.  You should use the same test framework you usually use for the language your Rest.li
+    client is written in. For the Python test suite, we use pytest. 
+
+4.  In your test framework, load the manifest.json file into a convenient in-memory representation before running the tests.
+
+    In Python, a Manifest class is generated from the .pdsc provided by the test suite. 
+    The ```manifest.json``` file is loaded into a Manifest representation that is used by the Python test suite.
+
+5.  Based on your test framework, decide how you want to use your Manifest representation to drive your automated tests. 
+
+    For example, consider json tests in the Python test suite, shown below. We get the list of manifest tests from the
+    Manifest representation in json_test_data(). Then we use the pytest @parametrize decorator to pass the test names
+    and data files into the function test_json(), which contains the testing logic. 
+    ```python
+    def json_test_data():
+        """Data provider for json_test"""
+        results = []
+        for test in manifest.get_json_test_data():
+            results.append(test)
+        return results
+    
+    
+    json_data = json_test_data()
+    
+    
+    @pytest.mark.parametrize("json_test", json_test_data(), ids=[test.data_ for test in json_data])
+    def test_json(json_test):
+        """Tests for correct serialization/deserialization of provided 'data' files"""
+        ...
+    ```   
+ 
+6.  Set up utility methods in your new language. These methods should provide a way to read in the files referenced by
+    manifest.json. You also need methods to compare and load flat HTTP requests and responses from the test suite's
+    provided test data. 
+ 
+    In the Java TestNG suite, these utility methods are part of a test suite base class that is extended by the classes
+    containing actual tests. The Python pytest suite has no subclass/superclass structure because it makes pytest harder to use.
+ 
+Now your test suite is set up to generate code and read language-independent files from the Rest.li Test Suite
+Specification. 
+
+### Writing Tests with the Provided Test Data
+The next step is to implement tests using the spec's guidelines and provided test data. 
+This will be a combination of manual assertions and data driven tests specified by ```manifest.json```.
+#### Data Driven Tests
+The tests are driven by ```manifest.json``` and cover three main categories: 
+* JSON: serialization/deserialization of JSON
+* schema: data template generation from schema
+* wire protocol: building HTTP requests and decoding Rest.li responses from HTTP responses
+
+##### JSON tests — driven by 'jsonTestData' and .json files 
```

Comment:
Heading can be simpler, just "JSON Tests", "Schema Tests", "Wire Protocol Tests" should be fine. Those files are already explained in overview.